src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -191,197 +371,190 @@ private ExecutorInfo createExecutor(String taskIdName, String nodeName, String e
-        liveState.removeStagingTask(status.getTaskId());
-        liveState.updateTask(status);
-
-        if (status.getTaskId().getValue().contains(HDFSConstants.NAME_NODE_TASKID)) {
-          if (liveState.getNameNodes().size() == HDFSConstants.TOTAL_NAME_NODES) {
-            //Finished initializing cluster after both name nodes are initialized
-            initializingCluster = false;
-          } else {
-            //Activate secondary name node after first name node is activated
-            for (TaskID taskId : liveState.getStagingTasks()) {
-              if (taskId.getValue().contains(HDFSConstants.NAME_NODE_TASKID)) {
-                sendMessageTo(driver, taskId, HDFSConstants.NAME_NODE_BOOTSTRAP_MESSAGE);
-                break;
-              }
-            }
-          }
-        } else if (status.getTaskId().getValue().contains(HDFSConstants.JOURNAL_NODE_ID) &&
-              (liveState.getJournalNodes().size() ==
-              (HDFSConstants.TOTAL_NAME_NODES + conf.getJournalNodeCount()))) {
-            //Activate primary name node after all journal nodes are activated
-            for (TaskID taskId : liveState.getStagingTasks()) {
-              if (taskId.getValue().contains(HDFSConstants.NAME_NODE_TASKID)) {
-                sendMessageTo(driver, taskId, HDFSConstants.NAME_NODE_INIT_MESSAGE);
-                break;
-              }
-            }
-         }
-      }
+  private boolean isStagingState(TaskStatus taskStatus) {
+    return (taskStatus.getState().equals(TaskState.TASK_STAGING));
   }
 
-  @Override
-  public void run() {
-    FrameworkInfo.Builder frameworkInfo = FrameworkInfo.newBuilder()
-        .setName("HDFS " + conf.getClusterName())
-        .setFailoverTimeout(conf.getFailoverTimeout())
-        .setUser(conf.getHdfsUser())
-        .setRole(conf.getHdfsRole())
-        .setCheckpoint(true);
+  private void reloadConfigsOnAllRunningTasks(SchedulerDriver driver) {
+    for (Protos.TaskStatus taskStatus : liveState.getRunningTasks().values()) {
+      sendMessageTo(driver, taskStatus.getTaskId(), taskStatus.getSlaveId(),
+          HDFSConstants.RELOAD_CONFIG);
+    }
+  }
 
-    try {
-      FrameworkID frameworkID = persistentState.getFrameworkID();
-      if (frameworkID != null) {
-        frameworkInfo.setId(frameworkID);
+  private void correctCurrentPhase() {
+    if (liveState.getJournalNodeSize() < conf.getJournalNodeCount()) {
+      liveState.transitionTo(AcquisitionPhase.JOURNAL_NODES);
+    } else if (liveState.getNameNodeSize() < HDFSConstants.TOTAL_NAME_NODES) {
+      liveState.transitionTo(AcquisitionPhase.START_NAME_NODES);
+    } else if (liveState.getNameNodeSize() == HDFSConstants.TOTAL_NAME_NODES) {
+      if (liveState.isNameNode1Initialized()
+          && liveState.isNameNode2Initialized()) {
+        liveState.transitionTo(AcquisitionPhase.DATA_NODES);
+      } else {
+        liveState.transitionTo(AcquisitionPhase.FORMAT_NAME_NODES);
```

Comment:
I would rather see this logic reversed, so the states match the execution order: if (!isNN1Init || !isNN2Init) { FORMAT_NN } else { DN }